tests/php/test-amp-layout-sanitizer.php

```diff
@@ -83,6 +83,31 @@ public function get_body_data() {
 				'<amp-img src="foo.jpg" style="height:100%;" width="100%"></amp-img>',
 				'<amp-img src="foo.jpg" layout="fill"></amp-img>',
 			],
+
+			'inside_amp_mustache_template_simple_text'    => [
+				'<template type="amp-mustache"><span>Hello, {{first_name}}!</span></template>',
+			],
+
+			'inside_amp_mustache_template_tag'            => [
+				'<template type="amp-mustache"><amp-img src="{{url}}" layout="intrinsic" srcset="{{foo}}" width="{{example_width}}" height="{{example_height}}" alt="This is for {{title}}"></amp-img></template>',
+			],
+
+			'inside_amp_mustache_script_tag'              => [
+				'<script template="amp-mustache" type="text/plain"><amp-img src="{{url}}" layout="intrinsic" srcset="{{foo}}" width="{{example_width}}" height="{{example_height}}" alt="This is for {{title}}"></amp-img></script>',
+			],
+
+			'inside_amp_mustache_template_responsive'     => [
+				'<template type="amp-mustache"><div><div><div><amp-img src="{{url}}" layout="responsive" srcset="{{foo}}" width="{{example_width}}" height="{{example_height}}" alt="This is for {{title}}"></amp-img></div></div></div></template>',
+			],
+
+			'inside_amp_mustache_template_sizes'          => [
+				'<template type="amp-mustache"><div><div><div><amp-img src="https://example.com/image.jpg" sizes="{{sizes}}" layout="responsive" width="200" height="400" alt="Lake vacation"></amp-img></div></div></div></template>',
```

Comment:
I suggest having multiple `amp-img` elements in here somewhat like:

```html
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="bad"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="{{layout}}" width="bad" height="bad"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="{{width}}" height="bad"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="{{height}}"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="bad" heights="{{height}}"></amp-img></div></div></div></template>
	<template type="amp-mustache"><div><div><div><amp-img src="/img1.png" layout="bad" width="bad" height="bad" heights="bad" sizes="{{sizes}}"></amp-img></div></div></div></template>
```

Where the first one is invalid but the others confirm all at once that the various attributes are accounted for.

Reply:
https://github.com/ampproject/amp-wp/pull/4583/commits/669285ea1982a9be1a62881da46bed7e2a431bd7 adds something similar to this.

Reply:
@kienstra Does testing those actually make the most sense in the `AMP_Layout_Sanitizer_Test`? Wouldn't it make better sense to test 

 I realize that the above sample should probably rather (or in addition) be added to the tag-and-attribute tests because this is where the mustache template vars are being checked, right? When doing so, it shows a difference of behavior from the AMP Validator. See 891c108651faf70d49c216e477de3ae7734fe395.

Now, it could be argued that the AMP plugin is actually doing something better because it is removing the attributes that could never be valid even when Mustache template handling is being performed. If that is the case, we should codify the behavior in this `mustache_templates_with_variable_attrs` test case to show how it behaves.

Reply:
Yeah, that test above isn't very useful. It's hard to get a failing test there, as the layout sanitizer [mainly deals with](https://github.com/ampproject/amp-wp/blob/9e4e491a5706efcb1d9f934b627ffeb8744ebe83/includes/sanitizers/class-amp-layout-sanitizer.php#L90-L100) `layout="fill"` and `layout="fixed-height"`.

Reply:
With your https://github.com/ampproject/amp-wp/pull/4583/commits/891c108651faf70d49c216e477de3ae7734fe395, I could revert the [test above](https://github.com/ampproject/amp-wp/pull/4583/files#diff-bf82ae038f1f6952929eaa9d4ae9710aR103) if you'd like.